### PR TITLE
Require CMake 3.14

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Require CMake 3.14 and remove workaround for older versions.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)
- Build related changes

### Does this introduce a breaking change?

- Yes

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
Okay to test

**Comment by prckent** [comment]
Cmake versions in the ALCF need to be increased.

**Comment by prckent** [comment]
Since this might cause software updates to be required, for the record I'll note that cmake 3.14.0 was released in March 2019. This exactly falls at the end of our 2 year support policy and will be even older by the next release. Since we need to benefit from new features and avoid ever-more complications from maintaining workarounds for older versions I consider this a very reasonable update.

**Comment by prckent** [comment]
@ye-luo Recognizing that it is a very busy week, any comments on a timeline for updating the ALCF CI cmake versions?

Here I am doing the second go around of software stack update for the nightlies on nitrogen and sulfur. Currently backing off using gcc 10.3 due to some configuration issues, and might have to back off to an older spack version. These updates aren't related to cmake but I usually try to update all the software versions when running an update. Once fixed the "old" cmake version that we test will be 3.14.0. The "new" cmake version was already new enough of course.

**Comment by ye-luo** [comment]
> @ye-luo Recognizing that it is a very busy week, any comments on a timeline for updating the ALCF CI cmake versions?
> 
> Here I am doing the second go around of software stack update for the nightlies on nitrogen and sulfur. Currently backing off using gcc 10.3 due to some configuration issues, and might have to back off to an older spack version. These updates aren't related to cmake but I usually try to update all the software versions when running an update. Once fixed the "old" cmake version that we test will be 3.14.0. The "new" cmake version was already new enough of course.

I actually got a 3.17 built on one CI node and need testing so hopefully finished tonight if not ealier. The other node has 3.14 already so I will use it directly. Let me know when your update completes, the merge should only happen after updates on our both sides are completed.

**Comment by prckent** [comment]
Should be good for nightlies.

**Comment by ye-luo** [comment]
Test this please

File: CMake/macros.cmake [hunk 1]
```diff
@@ -19,26 +19,17 @@ INCLUDE("${PROJECT_SOURCE_DIR}/CMake/test_labels.cmake")
 
 # Function to copy a directory
 FUNCTION( COPY_DIRECTORY SRC_DIR DST_DIR )
-    EXECUTE_PROCESS( COMMAND ${CMAKE_COMMAND} -E copy_directory "${SRC_DIR}" "${DST_DIR}" )
+    file(COPY ${SRC_DIR} DESTINATION ${DST_DIR})
```
Comment (ye-luo) [inline CMake/macros.cmake:22]: This changed behavior. 
COPY_DIRECTORY(folder_a folder_b)
is expect to produce a new folder_b not folder_b/folder_a

**Pull request merged**